components/brave_wallet/browser/ethereum_keyring_unittest.cc

```diff
@@ -260,4 +264,85 @@ TEST(EthereumKeyringUnitTest, GetPublicKeyFromX25519_XSalsa20_Poly1305) {
       "", &public_encryption_key));
 }
 
+TEST(EthereumKeyringUnitTest, DecryptCipherFromX25519_XSalsa20_Poly1305) {
+  EthereumKeyring keyring;
+  std::unique_ptr<std::vector<uint8_t>> seed = MnemonicToSeed(kMnemonic, "");
+  ASSERT_TRUE(seed);
+  keyring.ConstructRootHDKey(*seed, "m/44'/60'/0'/0");
+
+  // pub encryption key:
+  // "eui9/fqCHT7aSUkKK9eooQFnOCD9COK9Mi1ZtOxIj2A="
+  keyring.AddAccounts(1);
+
+  // {
+  //   version: 'x25519-xsalsa20-poly1305',
+  //   nonce: '2forT3nCPBRye9DeM1QSuWN8WvGZVpXN',
+  //   ephemPublicKey: 'mjgeR52JCGDl9336uQwN29qkbYR0WLKCginBPDFtyXY=',
+  //   ciphertext: 'e3xpKp2vGmpd9ecOjRJ6xltVr0Feo+Pm/C/gTMCyDg=='
```

Comment:
we also need to test `DecryptCipherFromX25519_XSalsa20_Poly1305` when 
1. plaintext is empty
2. plaintext is just one byte

maybe it is easier to test this scenario in hd_key unittest

Reply:
Added here 86beb4a089ad623aeb819b4a14f13a08fa112b76